### PR TITLE
Add rename command

### PR DESCRIPTION
**Pull request by zhouhao3**
Add rename command to rename a local container.

Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

**Comment by nalind** [comment]
This is probably going to make things confusing, because historically we've based our copy of the container's name on the name that we initially set using the storage library, and we're probably not being careful about one or the other changing.

Any chance we could make `Rename()` a method of the builder that updates the `Container` field and also calls the builder's `store.SetNames()` method to update the names that the storage library has on file?  We could also consider resetting `Container` based on the value that the storage library has as a part of opening the builder, which would keep us from being surprised when someone else changed the storage library's set of names for the container.  What do you think?

**Comment by q384566678** [comment]
```
➜  buildah git:(add-rename) sudo buildah containers
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
828e0b099441     *     3fd9065eaf02 docker.io/library/alpine:latest  alpine-working-container
➜  buildah git:(add-rename) sudo ./buildah rename alpine-working-container test-container
➜  buildah git:(add-rename) sudo ./buildah containers                                    
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
828e0b099441     *     3fd9065eaf02 docker.io/library/alpine:latest  alpine-working-container
➜  buildah git:(add-rename) sudo ./buildah run alpine-working-container bash
error reading build container "alpine-working-container": error reading build container: container not known
➜  buildah git:(add-rename) sudo ./buildah containers --all                 
CONTAINER ID  BUILDER  IMAGE ID     IMAGE NAME                       CONTAINER NAME
828e0b099441     *     3fd9065eaf02 docker.io/library/alpine:latest  test-container
```

Updated.  After a change, Because the structure of the call is different, only the --all option displays the name of the container that has been properly modified.

**Comment by q384566678** [comment]
@nalind @rhatdan @TomSweeneyRedHat PTAL

File: cmd/buildah/rename.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		ArgsUsage:      "CONTAINER-NAME-OR-ID CONTAINER-NAME",
+		SkipArgReorder: true,
+	}
+)
+
+func renameCmd(c *cli.Context) error {
+	var builder *buildah.Builder
+
+	args := c.Args()
+	if len(args) < 2 {
```
Comment (rhatdan) [inline cmd/buildah/rename.go]: Nit: Could simplify this code to say
if len(args) == 2 {
          return errors.Errorf("container and it's new name must be specified'")
}

File: cmd/buildah/rename.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	builder, err = openBuilder(getContext(), store, name)
+	if err != nil {
+		return errors.Wrapf(err, "error reading build container %q", name)
+	}
+
+	oldName := builder.Container
+	if oldName == newName {
+		return errors.Errorf("Renaming a container with the same name as its current name")
```
Comment (rhatdan) [inline cmd/buildah/rename.go]: Errors should be lower case.

**Comment by rhatdan** [comment]
You also need to update buildah.md man page.

**Comment by rhatdan** [comment]
Also add command to README.md

File: cmd/buildah/rename.go (hunk 1 above)
Comment (TomSweeneyRedHat) [reply cmd/buildah/rename.go]: s/b '!=2` no?  Or are the cold meds kicking in for me?

File: cmd/buildah/rename.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	err = store.SetNames(builder.ContainerID, []string{newName})
+	if err != nil {
+		return errors.Wrapf(err, "error set name %q to container %q", newName, oldName)
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/rename.go]: suggest:  "error renaming container %q to the name %q", oldName, newName

**Comment by q384566678** [comment]
Rebased.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+
@q384566678 I am not sure this can be done for podman.  @mheon What do you think?

**Comment by mheon** [comment]
@rhatdan Not easily, Podman container configs are as immutable as I could get away with, so it would require major changes to allow name changes.

**Comment by rhatdan** [comment]
Ok I guess we could put this at the bottom of the barrell of issues.

**Pull request closed without merging**